src/include/stir/numerics/erfMapping.h

```diff
@@ -0,0 +1,102 @@
+/*
+    Copyright (C) 2022, UCL
+    This file is part of STIR.
+
+    SPDX-License-Identifier: Apache-2.0
+
+    See STIR/LICENSE.txt for details
+*/
+/*!
+  \file
+  \ingroup erfMapping
+  \brief Implementation of a erf mapping
+
+  \author Robert Twyman
+*/
+
+//#include "stir/modulo.h"
+//#include "stir/common.h"
+//#include "stir/numerics/erf.h"
+
+#ifndef ERFMAPPING_H
+#define ERFMAPPING_H
+
+START_NAMESPACE_STIR
+
+
+//! erfMapping is a class to save values of the erf function
+class erfMapping
+{
+private:
+
+  struct MappingTable
+  {
+    double x;
+    double y;
+  };
+
+  double interpolate( const std::vector<MappingTable> &data, double MappingTable::*x, double MappingTable::*y, double xValue ) const
+  {
+    auto ip = lower_bound( data.begin(), data.end(), xValue, [ x ]( MappingTable a, double b ){ return a.*x < b; } );
+
+    // No extrapolation beyond data limits
+    if ( ip == data.end  () )
+    {
+      return data.back ().*y;
+    }
+    else if ( ip == data.begin() )
+    {
+      return data.front().*y;
+    }
+
+    // Otherwise, linear interpolation
+    auto im = ip - 1;
+    double slope = ( (*ip).*y - (*im).*y ) / ( (*ip).*x - (*im).*x );      // dY/dX
+    return (*im).*y + slope * ( xValue - (*im).*x );                       // Y0 + (dY/dX) * ( X - X0 )
+  }
+
+  int num_entries;
+  double lower_range;
+  double upper_range;
+  std::vector<MappingTable> map;
+
+public:
+  erfMapping ()       //constructor 1 with no arguments
+  {
+    this->num_entries = 1;
+    this->lower_range = 0;
+    this->upper_range = 5;
+  }
+  erfMapping(int n)    //constructor 2 with one argument
+  {
+    this->num_entries = n;
+    this->lower_range = 0;
+    this->upper_range = 5;
+  }
```

Comment:
probably these make no sense, but if they do, you should have just one with default args. Also add `explicit` such that the 1-arg constructor isn't used for conversion.